src/components/FiatOnrampModal/util.ts

```diff
@@ -0,0 +1,75 @@
+import { ChainId, WETH9 } from '@uniswap/sdk-core'
+import { MATIC, USDC_ARBITRUM, USDC_MAINNET, USDC_OPTIMISM, USDC_POLYGON, USDT, WBTC } from 'constants/tokens'
+import { Chain } from 'graphql/data/__generated__/types-and-hooks'
+import { validateUrlChainParam } from 'graphql/data/util'
+import { isSameAddress } from 'utils/addresses'
+
+export const MOONPAY_SUPPORTED_CURRENCY_CODES = [
```

Comment:
you can convert this to a map of addresses and chains which would greatly reduce the amount of copy and pasted code needed in the `getDefaultCurrencyCode` function